Hyper emits opaque errors when streams have encountered errors and got shut down.

Seen in the release tests:

```
_______________ TestHyperActuallyWorks.test_abusing_nghttp2_org _______________

self = <test_release.TestHyperActuallyWorks object at 0x0505C5B0>

    def test_abusing_nghttp2_org(self):
        """
            This test function loads all of nghttp2.org's pages in parallel. This
            tests us against the most common open source HTTP/2 server
            implementation.
            """
        paths = [
            '/',
            '/blog/2014/04/27/how-dependency-based-prioritization-works/',
            '/blog/2014/04/25/http-slash-2-draft-12-update/',
            '/blog/2014/04/23/nghttp2-dot-org-now-installed-valid-ssl-slash-tls-certificate/',
            '/blog/2014/04/21/h2load-now-supports-spdy-in-clear-text/',
            '/blog/2014/04/18/nghttp2-dot-org-goes-live/',
            '/blog/archives/',
        ]

        c = HTTP20Connection('nghttp2.org', enable_push=True)

        # Make all the requests, then read the responses in a random order.
        stream_ids = [c.request('GET', path) for path in paths]
        random.shuffle(stream_ids)
>       responses = [c.get_response(i) for i in stream_ids]

test_release.py:42:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
hyper\http20\connection.py:186: in get_response
    stream = self._get_stream(stream_id)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <hyper.http20.connection.HTTP20Connection object at 0x0505C610>
stream_id = 13

    def _get_stream(self, stream_id):
>       return (self.streams[stream_id] if stream_id is not None
                else self.recent_stream)
E       KeyError: 13

hyper\http20\connection.py:170: KeyError
```

The actual error here is that the stream was reset. We should be able to provide a more useful error than that. We might want to piggyback on the error from #146.

--- a/hyper/http20/connection.py
+++ b/hyper/http20/connection.py
@@ -19,7 +19,7 @@ from ..compat import unicode, bytes
 from .stream import Stream
 from .response import HTTP20Response, HTTP20Push
 from .window import FlowControlManager
-from .exceptions import ConnectionError
+from .exceptions import ConnectionError, StreamResetError
 from . import errors
 
 import errno
@@ -186,8 +186,12 @@ class HTTP20Connection(object):
         return stream_id
 
     def _get_stream(self, stream_id):
-        return (self.streams[stream_id] if stream_id is not None
-                else self.recent_stream)
+        if stream_id is None:
+            return self.recent_stream
+        elif stream_id in self.reset_streams or stream_id not in self.streams:
+            raise StreamResetError("Stream forcefully closed")
+        else:
+            return self.streams[stream_id]
 
     def get_response(self, stream_id=None):
         """

--- a/hyper/http20/stream.py
+++ b/hyper/http20/stream.py
@@ -14,7 +14,6 @@ Each stream is identified by a monotonically increasing integer, assigned to
 the stream by the endpoint that initiated the stream.
 """
 from ..common.headers import HTTPHeaderMap
-from .exceptions import StreamResetError
 from .util import h2_safe_headers
 import logging
 
@@ -201,7 +200,7 @@ class Stream(object):
         Stream forcefully reset.
         """
         self.remote_closed = True
-        raise StreamResetError("Stream forcefully closed")
+        self._close_cb(self.stream_id)
 
     def get_headers(self):
         """

--- a/test/test_integration.py
+++ b/test/test_integration.py
@@ -678,7 +678,7 @@ class TestHyperIntegration(SocketLevelTest):
         conn._recv_cb()
 
         # However, attempting to get the response should.
-        with pytest.raises(KeyError):
+        with pytest.raises(StreamResetError):
             conn.get_response(stream_id)
 
         # Awesome, we're done now.
@@ -716,13 +716,13 @@ class TestHyperIntegration(SocketLevelTest):
         conn = self.get_connection()
         conn.request('GET', '/')
 
-        # Now, eat the RstStream frames. The first one throws a
-        # StreamResetError.
-        with pytest.raises(StreamResetError):
-            conn._single_read()
-
-        # The next should throw no exception.
+        # Now, eat the Rst frames. These should not cause an exception.
         conn._single_read()
+        conn._single_read()
+
+        # However, attempting to get the response should.
+        with pytest.raises(StreamResetError):
+            conn.get_response(1)
 
         assert conn.reset_streams == set([1])
 

